mlir/lib/Dialect/Linalg/Transforms/Vectorization.cpp

```diff
@@ -890,11 +907,10 @@ static bool isContiguousLoadIdx(LinalgOp &linalgOp, Value &val,
   Operation *defOp = val.getDefiningOp();
   assert(defOp && "This is neither a block argument nor an operation result");
 
-  // Given the assumption on the loop ranges above, only the trailing loop
-  // index is not constant.
-  auto trailingLoopDim = linalgOp.getStaticLoopRanges().size() - 1;
+  auto nonUnitLoopDim = getNonUnitLoopDim(linalgOp);
```

Comment:
[nit] It's worth adding a comment that given the assumption above (encoded in `assert`), we expect only 1 non-unit loop dim.